pkg/allocator/cache.go

```diff
@@ -122,6 +129,10 @@ func (c *cache) OnAdd(id idpool.ID, key AllocatorKey) {
 	}
 	c.allocator.idPool.Remove(id)
 
+	c.emitChange(AllocatorChange{Kind: AllocatorChangeUpsert, ID: id, Key: key})
+
+	// TODO: Remove sendEvent in favour of the observable. Might need
+	// to support modify? Or just Upsert?
```

Comment:
Just to double-check., is this supposed to be tackled in a followup PR? Even in this case, it might be worth to decide whether to split the UpsertEvent into create and update to avoid the need for a subsequent refactoring (e.g., [here](https://github.com/cilium/cilium/blob/main/pkg/identity/cache/cache.go#L153-L160) it seems that update events are ignored).

Reply:
nothing planned so far. But anyway - would be easy to refactor into any direction as long as there aren't many usages of this observable. `Upsert` makes it a 1:1 replacement when replacing `Resource[CiliumIdentity]

Reply:
I see. The typical issue is only that then usages pile up and refactoring becomes more onerous, but I agree that it is not something that needs to be addressed right now.

Reply:
I believe `TODO`s are frowned upon, GH issues are preferred

Reply:
i removed the comment. will discuss with @joamaki whether we want to create a GH issue or not